af/fastpath/fastpath/core.py

```diff
@@ -1213,6 +1213,22 @@ def score_dns_consistency(msm) -> dict:
     return scores
 
 
+def score_signal(msm) -> dict:
+    """Calculates measurement scoring for Signal test
+    Returns a scores dict
+    """
+    scores = {f"blocking_{l}": 0.0 for l in LOCALITY_VALS}
+    tk = msm.get("test_keys", {})
+    if tk.get("failed_operation", True) or tk.get("failure", True):
+        scores["accuracy"] = 0.0
+
+    for c in tk.get("tcp_connect", []):
+        if c.get("status", {}).get("success", False) == False:
+            scores["blocking_general"] = 1.0
+
+    return scores
```

Comment:
You should be reading the keys:
```
{
    "signal_backend_failure": "FAILURE STRING",
    "signal_backend_status": "blocked"
}
```
See: https://github.com/ooni/spec/blob/master/nettests/ts-029-signal.md